Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+separate failure domain just for the third monitor node. As such, our hypothetical data center has
+two rooms and one failure domain, with the following configuration:
+* The failure domain is small and can only to be used for the third Ceph MON;
+  it does not have space for storage nodes.
+* Eight Ceph OSD nodes provide a good amount of data safety without requiring too many nodes.
+* These eight nodes should be equally separated — four to each data center room.
+* We can imagine that the four nodes are separated in each room into two racks.
+* In the event of a MON node failure, let's decide we want to be able to run MONs on each rack
+
+The data center looks as follows:
+
+![Example Datacenter](media/ceph-bp-example-datacenter.png)
+
+Now we will dig a little deeper and talk about the actual disks used for Rook and Ceph storage. To
+ensure we are following known Ceph best practices for this data center setup, ensure that MON
+storage goes on the SSDs. We'll discuss this a little more later. Because each rack should be able
+to run a Ceph MON, one of the nodes in each rack will have an SSD that is usable for MON storage.
+Additionally, all nodes in all racks (except in the failure domain) will have disks for OSD storage.
+This will look like the following:
+
+![Example Datacenter - Disks](media/ceph-bp-example-datacenter-disks.png)
+
+> **IMPORTANT:** Refer to these diagrams when we discuss the example data center below.
+
+
+## Introduction
+
+Ceph and Kubernetes both have their own well-known and established best practices. Rook bridges the
+gap between Ceph and Kubernetes, putting it in a unique domain with its own best practices to
+follow. Therefore, this document specifically covers best practices for running Ceph on Kubernetes
+with Rook. Because Rook augments on top of Kubernetes, it has different ways of meeting both Ceph
+and Kubernetes best practices. This is in comparison to the bare metal versions of each. Out of the
+box, Rook is predominantly a default Ceph cluster. The Ceph cluster needs tuning to meet user
+workloads, and Rook does not absolve the user from planning out their production storage cluster
+beforehand.
+
+For the purpose of this document, we will consider two simplified use cases to help us make informed
+decisions about Rook and Ceph:
+ * Co-located: User applications co-exist on nodes running Ceph
+ * Disaggregated: Ceph nodes are totally separated from user applications
+
+
+## General Best Practices
+
+This chapter provides an outline of a series of generalized recommendations for best practices:
+* Ceph monitors are more stable on fast storage (SSD-class or better) according to Ceph best
+  practices. In Rook, this means that the `dataDirHostPath` location in the `cluster.yaml` should be
+  backed by SSD or better on MON nodes.
+* Raise the Rook log level to `DEBUG` for initial deployment and for upgrades, as it will help with
```

Comment:
I'm willing to discuss more, but... this is at least still the recommendation I would give to anyone. I don't have specific examples, but I definitely notice that the INFO logs generally aren't enough for debugging issues I have seen on Slack or that I get internally. While this might be an uncomfortable suggestion, I think it is the right one.

Reply:
Well, let's strive to fix it. But if you like the debug logs, you must like more detailed logs than the average person. :)